src/widget/invlerp_range_finder.ts

```diff
@@ -0,0 +1,235 @@
+    return this.autoRangeData.finishedLerpRange;
+  }
+
+  private wasInputInverted() {
+    const { range } = this.parent.trackable.value;
+    return range !== undefined && range[0] > range[1];
+  }
+
+  autoComputeRange(minPercentile: number, maxPercentile: number) {
+    if (!this.autoRangeData.autoComputeInProgress) {
+      const { autoRangeData } = this;
+      const { dataType, display } = this.parent;
+
+      // Reset the auto-compute state
+      autoRangeData.inputPercentileBounds = [minPercentile, maxPercentile];
+      this.resetAutoRangeData(autoRangeData);
+      autoRangeData.invertedInitialRange = this.wasInputInverted();
+      display.force3DHistogramForAutoRange = true;
+
+      // Create a large range to search over
+      // It's easier to contract the range than to expand it
+      const maxRange =
+        dataType === DataType.FLOAT32
+          ? ([-65536, 65536] as [number, number])
+          : defaultDataTypeRange[dataType];
+      this.setTrackableValue(maxRange, maxRange);  
+      display.scheduleRedraw();
+    }
+  }
+
+  setTrackableValue(range: DataTypeInterval, window: DataTypeInterval) {
+    const hasRange = this.parent.trackable.value.range !== undefined;
+
+    const ensureWindowBoundsNotEqual = (window: DataTypeInterval) => {
+      if (dataTypeCompare(window[0], window[1]) === 0) {
+        return defaultDataTypeRange[this.parent.dataType];
+      }
+      return window;
+    };
+
+    if (hasRange) {
+      this.parent.trackable.value = {
+        ...this.parent.trackable.value,
+        range,
+        window,
+      };
+    } else {
+      this.parent.trackable.value = {
+        ...this.parent.trackable.value,
+        window: ensureWindowBoundsNotEqual(window),
+      };
+    }
+  }
+
+  public maybeAutoComputeRange() {
+    if (!this.autoRangeData.autoComputeInProgress) {
+      this.parent.display.force3DHistogramForAutoRange = false;
+      return;
+    }
+    const { autoRangeData } = this;
+    const {
+      trackable,
+      dataType,
+      display,
+      histogramSpecifications,
+      histogramIndex,
+    } = this.parent;
+    const gl = display.gl;
+    let { range } = trackable.value;
+    if (range === undefined) {
+      range = trackable.value.window;
+    }
+
+    // Read the histogram from the GPU and compute new range based on this
+    const frameBuffer =
+      histogramSpecifications.getFramebuffers(gl)[histogramIndex];
+    frameBuffer.bind(256, 1);
+    const empiricalCdf = copyHistogramToCPU(gl);
+    const { range: newRange, window: newWindow } =
+      computePercentilesFromEmpiricalHistogram(
+        empiricalCdf,
+        autoRangeData.inputPercentileBounds[0],
+        autoRangeData.inputPercentileBounds[1],
+        range,
+        dataType,
+      );
+
+    // If the range remains constant over two iterations
+    // or if we've exceeded the maximum number of iterations, stop
+    const foundRange =
+      autoRangeData.lastComputedLerpRange !== null &&
```

Comment:
better check probably to keep all of the computed ranges, and stop if we ever find the same again (bounce possible)